src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,124 @@
+package duke.tasks;
+
+import duke.dukeexception.DateTimeFormatException;
+
+/**
+ * The base class for Task.
+ */
+public class Task {
+    protected static Integer totalNumber = 0;
+    protected String name;
+    protected Integer index;
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task object.
+     * @param name The description of task.
+     */
+    public Task(String name) {
+        this.name = name;
+        totalNumber++;
+        this.index = totalNumber;
+        this.isDone = false;
+    }
+
+    /**
+     * Helper function to call while adding a new task.
+     * @param name the new task description.
+     * @return the duke.Tasks.Task class being created.
+     */
+    public static Task addTask(String name) throws DateTimeFormatException {
+        Task newTask = new Task(name);
+        System.out.println("       " + newTask.printSelf());
+        return newTask;
+    }
+
+    /**
+     * Helper function to call while deleting a task, to keep track of the total number of tasks.
+     * @param task The task being deleted.
+     */
+    public static void removeTask(Task task) throws DateTimeFormatException {
+        totalNumber--;
+        System.out.println("       " + task.printSelf());
+    }
+
+    /**
+     * Update the index of a particular task to be index -1 after removal.
+     */
+    public void updateRemoval() {
```

Comment:
updateRemoval sounds more like a noun, is it possible to make it sound more like a verb?